### ISSUE TITLE
Ignore Windows toast messages while in sleep mode

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#7626]
### Steps to reproduce:
1. GO to a secure desktop, NVDA should sleep
2. Trigger a Windows 10 toast. On my system (HP Probook), a toast is triggered when a device is plugged into the head phone jack. Other toasts that can be fired include messages about actions required to connect to a wifi network.

### Expected behavior:
The user copy of NVDA remains silent, as it is in sleep mode.

### Actual behavior:
The user copy of NVDA speaks the incoming toast messages. Since you are on the secure desktop, you are unable to silence these messages.

### System configuration:
NVDA next-14463,7dca66c4, Windows 10 creators update

**Comment by derekriemer** [comment]
Agree that not being able to silence those is a bug, however, they do I think? visually show up.

**Comment by feerrenrut** [comment]
I don't consider it a bug, when I plug headphones into my system on the windows lock screen I see a notification. Why shouldn't NVDA make you aware of this information?

**Comment by derekriemer** [comment]
it's more that I can't tell NVDA to shut up after it starts going blah blah blah.

**Comment by leonardder** [comment]
@feerrenrut: Do you also see this notification when you are at a logon prompt? Note that on the Windows 10 lock screen itself still talks with the user instance. The problem applies to toast notification spoken by the user instance when the system instance has foreground.

**Comment by feerrenrut** [comment]
Ah, I tested this again. The messages are shown visually on the lock screen, but not at the logon prompt. I understand what you mean about not being able to silence the messages as well, you can actually get NVDA talking twice which is pretty confusing to deal with.  We should try to match the visual behaviour, speak the messages when on the lock screen, but not when on the logon prompt.

I'm going to try to clarify the title and description of this issue a little bit.

**Comment by leonardder** [comment]
Sorry for being a bit vague here.

I still belief this can be generalised to silencing toast messages, not specifically when on the logon screen, but in general when sleep mode is on. Note that API messages also come through while sleep mode is on.

**Comment by Adriani90** [comment]
It seems this occurs also for speech sent through controller client, not only for toast notificatoins. See also #2163.

**Comment by Adriani90** [comment]
This is also reproducible on Windows 11 24 H2 with NVDA 2025.1.2.

**Comment by Simon818** [comment]
This behaviour is definitely a bug while at the logon screen. I'm of the opinion that nothing whatsoever should be spoken by the user-facing copy of NVDA while that user's session is not active. This exposes information that would visually be hidden on the login screen, such as WhatsApp notifications being read in full, accessibility announcements sent to the controller, etc. I suspect but have not confirmed that even when a second user is logged in, the first user's session is still set to actively speak. There's not really a way around this except for remembering to turn off speech before locking the machine.
